tests/unittest/auto_deploy/singlegpu/shim/test_llm_config.py

```
@@ -179,6 +180,33 @@ def test_parallel_config_validation(parallel_field, invalid_value):
+            speculative_model="mtp-model",
+            use_mtp_vanilla=True,
+        ),
+    ],
+)
+def test_unsupported_speculative_config_validation(speculative_config):
+    """Test that AutoDeploy rejects unsupported speculative decoding modes."""
+    with pytest.raises(
+        ValueError,
+        match=(
+            "Currently, AutoDeploy only supports speculative decoding in "
+            "eagle3_one_model or mtp_eagle_one_model mode."
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)